thirdparty/patches/glog-dump-stacktrack.patch

```diff
@@ -0,0 +1,72 @@
+diff --git src/glog/logging.h.in src/glog/logging.h.in
+index 64bed1a..0ff073f 100644
+--- src/glog/logging.h.in
++++ src/glog/logging.h.in
+@@ -1676,6 +1676,9 @@ class GOOGLE_GLOG_DLL_DECL NullStreamFatal : public NullStream {
+ // words, stack traces of other threads won't be shown.
+ GOOGLE_GLOG_DLL_DECL void InstallFailureSignalHandler();
+ 
++// Providing stack trace function for upper level.
++GOOGLE_GLOG_DLL_DECL std::string GetStackTraceToString();
++
+ // Installs a function that is used for writing the failure dump.  "data"
+ // is the pointer to the beginning of a message to be written, and "size"
+ // is the size of the message.  You should not expect the data is
+diff --git src/logging.cc src/logging.cc
+index 733ce8c..e988eb9 100644
+--- src/logging.cc
++++ src/logging.cc
+@@ -2336,4 +2336,11 @@ void DisableLogCleaner() {
+   log_cleaner_enabled_ = false;
+ }
+ 
++std::string GetStackTraceToString() {
++  // Collect stacktrace from utilities.
++  std::string stacktrace;
++  DumpStackTraceToString(&stacktrace, 0);
```

Comment:
How does this work? Does it mean we won't dump any stacktrace? (depth 0)

Reply:
Depth means stack skip count. For example, we call f1->f2->f3 and full stack view is {CallStack, f1, f2, f3} with 0 depth. If the given depth is 1, stack sequence will be {f1, f2, f3}. In short,  no stack frame will be skipped if we use DumpStackTraceToString(&stacktrace, 0);

Reply:
BTW, max stack size is 32 by default.

Reply:
Ah, I see interesting. Can you write a comment about it?